### PR TITLE
Add build instruction in README, reorganize some sections

### PR DESCRIPTION
**Pull request by strk**
It's easier to find build instructions in there, and I also added
usage instructions

### CODE REVIEW

**Comment by lunny** [comment]
Maybe a link to the build docs is better since maybe the building process will be changed.

**Comment by techknowlogick** [comment]
LGTM

**Review by appleboy — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -31,24 +31,36 @@ This project has been
...
-## Notes
+## Building
 
-1. **YOU MUST READ THE [CONTRIBUTORS GUIDE](CONTRIBUTING.md) BEFORE STARTING TO WORK ON A PULL REQUEST.**
-2. If you have found a vulnerability in the project, please write privately to **security@gitea.io**. Thanks!
-3. If you're interested in using our APIs, we have experimental support with [documentation](https://godoc.org/code.gitea.io/sdk/gitea).
+From the root of the source tree, run:
 
-## Docs
+    make generate all
```
Comment (appleboy) [inline README.md:38]: maybe also add build from source documentation: https://docs.gitea.io/en-us/install-from-source/

**Comment by strk** [comment]
> maybe also add build from source documentation: https://docs.gitea.io/en-us/install-from-source/

Added with 135fdf97

**Pull request merged**